CMakeLists.txt

```diff
@@ -1262,7 +1262,11 @@ macro(add_test_prog prog)
                           event_core
                           event_extra
                           ${ARGN})
+    if(PTHREADS_AVAILABLE AND ${prog} STREQUAL "test-kq-collision")
+        target_link_libraries(${prog} event_pthreads)
```

Comment:
Again, I don't like this way, why not pass it explicitly only for tet-kq-collision?

Reply:
Oh, I guess I misunderstood your previous comment on this. Now I'm not sure I comprehend the "pass it explicitly only for tet-kq-collision" thoroughly, could you be more concrete?

Reply:
```
    foreach (TESTPROG ${TESTPROGS} test-dumpevents)
        add_test_prog(${TESTPROG})
    endforeach()
    if(PTHREADS_AVAILABLE)
        add_test_prog(test-kq-collision event_pthreads)
    endif()
```

And then you need to run it only if it was compiled

Reply:
It fails the CI after this change: https://github.com/libevent/libevent/actions/runs/8718766424/job/23916681856?pr=1582, https://github.com/libevent/libevent/actions/runs/8718766424/job/23916682713?pr=1582

Reply:
I was suggesting this:
>And then you need to run it only if it was compiled

I've addressed this by myself and also couple of warnings , and force kqueue backend for test-kq-collision.c